vslib/src/SwitchStateBase.cpp

```diff
@@ -629,6 +685,33 @@ sai_status_t SwitchStateBase::get(
     return final_status;
 }
 
+sai_status_t SwitchStateBase::get(
+        _In_ sai_object_type_t objectType,
+        _In_ const std::string &serializedObjectId,
+        _In_ uint32_t attr_count,
+        _Out_ sai_attribute_t *attr_list)
+{
+    SWSS_LOG_ENTER();
+
+    if (objectType == SAI_OBJECT_TYPE_MACSEC_SA)
+    {
+        return getMACsecSAAttr(
+            serializedObjectId,
+            attr_count,
+            attr_list);
+    }
+
+    if (objectType == SAI_OBJECT_TYPE_MACSEC)
+    {
+        return getMACsecAttr(
+            serializedObjectId,
+            attr_count,
+            attr_list);
+    }
+
+    return get_internal(objectType, serializedObjectId, attr_count, attr_list);
+}
```

Comment:
please remove get_inernal function, there are no special action needed for GET, those attributes SAI_MACSEC_ATTR_SCI_IN_INGRESS_MACSEC_ACL you are referring, are READ_ONLY and should be refressed in "status = refresh_read_only(meta, oid);" function internals

Reply:
SAI_MACSEC_SA_ATTR_XPN should be set during CREATE, since it's create and set attribute, so get will work fine for this attribute, same for SAI_MACSEC_SA_ATTR_MINIMUM_XPN, those attributes have default values, but they are not set, when object is created, probably you should set them on custom create function

Reply:
those 2 functions getMACsecSAAttr getMACsecAttr can be removed